overlord/snapstate/handlers.go

```diff
@@ -1280,6 +1276,70 @@ func daemonRestartReason(st *state.State, typ snap.Type) string {
 	return "Requested daemon restart (snapd snap)."
 }
 
+// maybeUndoRemodelBootChanges will check if an undo needs to update the
+// bootloader. This can happen if e.g. a new kernel gets installed. This
+// will switch the bootloader to the new kernel but if the change is later
+// undone we need to switch back to the kernel of the old model.
+func maybeUndoRemodelBootChanges(t *state.Task, undoInfo *snap.Info) error {
+	var oldModelStr string
+	err := t.Change().Get("old-model", &oldModelStr)
```

Comment:
we probably want  DeviceCtx to have a OldModel  accessor that works if ForRemodeling is true, that's quite easy to implement in devicestate I think

Reply:
I created https://github.com/snapcore/snapd/pull/7685 for this now.